pandas/core/generic.py

```diff
@@ -5134,12 +5134,15 @@ def tail(self: FrameOrSeries, n: int = 5) -> FrameOrSeries:
     @final
     def sample(
         self: FrameOrSeries,
-        n=None,
-        frac=None,
-        replace=False,
-        weights=None,
-        random_state=None,
-        axis=None,
+        n: Optional[int] = None,
+        frac: Optional[float] = None,
+        replace: bool_t = False,
+        weights: Union[str, np.ndarray] = None,
+        random_state: Optional[
+            Union[int, arraylike, np.random.BitGenerator, np.random.RandomState]
+        ] = None,
+        axis: Union[int, str] = None,
```

Comment:
There is already
```python
Axis = Union[str, int]
```
in `_typing.py`

Also, I guess `arraylike` was meant to be `ArrayLike`?

Reply:
Also this would need to be `Optional[Axis]` since can be `None`

Reply:
Thanks for the feedback, made the adjustments.